app/audit/ViewNotificationEvent.scala

```diff
@@ -41,3 +42,22 @@ object ViewNotificationEvent {
       )
     )
 }
+
+object ViewNotificationEventFailed {
+  def apply
+  (amlsRegistrationNumber: String, contactNumber: String, ex: HttpStatusException)
+  (implicit
+   hc: HeaderCarrier,
+   resW: Writes[NotificationResponse]
+  ): DataEvent =
+    DataEvent(
+      auditSource = AuditHelper.appName,
+      auditType = "viewNotificationEventFailed",
+      tags = hc.toAuditTags("Get Notification Failed", "N/A"),
+      detail = hc.toAuditDetails() ++ Map(
+        "amlsRegNo" -> amlsRegistrationNumber,
+        "contactNumber" -> contactNumber,
+        "reason" -> Json.toJson(ex.body.getOrElse("No body found")).toString
```

Comment:
Out of curiosity, why do we need `toString` ?

Reply:
The DataEvent takes params of type string. So if we remove the .toString() we encounter a compiler error;

found   : scala.collection.immutable.Map[String,Object]
 required: Map[String,String]

Reply:
👍